src/variants.js

```diff
@@ -55,89 +90,136 @@ export async function getVariant(name) {
 	return variants.find(variant => variant.name === name);
 }
 
-export const shortForms = /** @type {const} */ ({
-	'Red': 'r',
-	'Yellow': 'y',
-	'Green': 'g',
-	'Blue': 'b',
-	'Purple': 'p',
-	'Teal': 't',
-	'Black': 'k',
-	'Rainbow': 'm',
-	'White': 'w',
-	'Pink': 'i',
-	'Brown': 'n',
-	'Omni': 'o',
-	'Null': 'u',
-	'Prism': 'i'
-});
+export let shortForms = /** @type {string[]} */ (['r', 'y', 'g', 'b', 'p']);
+
+/**
+ * Edits shortForms to have the correct acryonyms.
+ * @param {string[]} suits
+ */
+export async function getShortForms(suits) {
+	const colors = await colors_promise;
+	const abbreviations = [];
+	for (const suitName of suits) {
+		if (['Black', 'Pink', 'Brown'].includes(suitName)) {
+			abbreviations.push(['k', 'i', 'n'][['Black', 'Pink', 'Brown'].indexOf(suitName)]);
+		} else {
+			const abbreviation = colors.find(color => color.name === suitName)?.abbreviation ?? suitName.charAt(0);
+			if (abbreviations.includes(abbreviation.toLowerCase())) {
+				for (const char of suitName) {
```

Comment:
This for-loop can be replaced with a more functional line:
```js
abbreviations.push(suitName.find(char => !abbreviations.includes(char)).toLowerCase());
```
(though I do wonder what'll happen when purple + prism + brown + black + pink becomes a thing)